pallets/api/src/nonfungibles/mod.rs

```diff
@@ -0,0 +1,591 @@
+		/// - `to` - The recipient account.
+		#[pallet::call_index(3)]
+		#[pallet::weight(NftsWeightInfoOf::<T>::transfer())]
+		pub fn transfer(
+			origin: OriginFor<T>,
+			collection: CollectionIdOf<T>,
+			item: ItemIdOf<T>,
+			to: AccountIdOf<T>,
+		) -> DispatchResult {
+			let from = ensure_signed(origin.clone())?;
+			NftsOf::<T>::transfer(origin, collection, item, T::Lookup::unlookup(to.clone()))?;
+			Self::deposit_event(Event::Transfer {
+				collection,
+				item,
+				from: Some(from),
+				to: Some(to),
+				price: None,
+			});
+			Ok(())
+		}
+
+		/// Approves `operator` to spend the collection item on behalf of the caller.
+		///
+		/// # Parameters
+		/// - `collection` - The collection of the item to approve for a delegated transfer.
+		/// - `item` - The item to approve for a delegated transfer.
+		/// - `operator` - The account that is allowed to spend the collection item.
+		/// - `approved` - The approval status of the collection item.
+		#[pallet::call_index(4)]
+		#[pallet::weight(
+            NftsWeightInfoOf::<T>::approve_transfer(item.is_some() as u32) +
+    		NftsWeightInfoOf::<T>::cancel_approval(item.is_some() as u32)
+        )]
+		pub fn approve(
+			origin: OriginFor<T>,
+			collection: CollectionIdOf<T>,
+			item: Option<ItemIdOf<T>>,
+			operator: AccountIdOf<T>,
+			approved: bool,
+		) -> DispatchResultWithPostInfo {
+			let owner = ensure_signed(origin.clone())?;
+			let weight = if approved {
+				NftsOf::<T>::approve_transfer(
+					origin,
+					collection,
+					item,
+					T::Lookup::unlookup(operator.clone()),
+					None,
+				)
+				.map_err(|e| {
+					e.with_weight(NftsWeightInfoOf::<T>::approve_transfer(item.is_some() as u32))
+				})?;
+				NftsWeightInfoOf::<T>::approve_transfer(item.is_some() as u32)
+			} else {
+				NftsOf::<T>::cancel_approval(
+					origin,
+					collection,
+					item,
+					T::Lookup::unlookup(operator.clone()),
+				)
+				.map_err(|e| {
+					e.with_weight(NftsWeightInfoOf::<T>::cancel_approval(item.is_some() as u32))
+				})?;
+				NftsWeightInfoOf::<T>::cancel_approval(item.is_some() as u32)
+			};
+			Self::deposit_event(Event::Approval { collection, item, operator, owner, approved });
+			Ok(Some(weight).into())
+		}
+
+		/// Issue a new collection of non-fungible items from a public origin.
+		///
+		/// # Parameters
+		/// - `admin` - The admin of this collection. The admin is the initial address of each
+		/// member of the collection's admin team.
+		/// - `config` - The configuration of the collection.
+		#[pallet::call_index(7)]
+		#[pallet::weight(NftsWeightInfoOf::<T>::create())]
+		pub fn create(
+			origin: OriginFor<T>,
+			admin: AccountIdOf<T>,
+			config: CollectionConfigFor<T>,
+		) -> DispatchResult {
+			let creator = ensure_signed(origin.clone())?;
+			// TODO: re-evaluate next collection id in nfts pallet. The `Incrementable` trait causes
+			//  issues for setting it to xcm's `Location`. This can easily be done differently.
```

Comment:
> TODO: re-evaluate next collection id in nfts pallet. The `Incrementable` trait cause issues for setting it to xcm's `Location`. This can easily be done differently.

Should this TODO be tackled in this PR?